lib/util/device-service-details.ts

```diff
@@ -110,26 +119,37 @@ export const generateCurrentServiceDetails = (
 		getSingleInstallSummary(gd),
 	);
 
-	const device = omit(rawDevice, [
-		'image_install',
-		'gateway_download',
-	]) as DeviceWithServiceDetails;
-
-	device.current_services = mapValues(
-		groupBy(installs, 'service_name'),
-		serviceContainers => {
-			return serviceContainers
-				.map(container => {
-					return omit(container, 'service_name') as Omit<
-						typeof container,
-						'service_name'
-					>;
-				})
-				.sort((a, b) => {
-					return b.install_date.localeCompare(a.install_date);
-				});
-		},
+	// prefer over omit for performance reasons
+	delete rawDevice.image_install;
+	delete rawDevice.gateway_download;
+
+	const device = (rawDevice as Device) as DeviceWithServiceDetails;
+
+	// Essentially a groupBy(installs, 'service_name')
+	// but try making it a bit faster for the sake of large fleets
+	const currentServicesGroupedByName = installs.reduce<
```

Comment:
I'd switch this block to
```ts
	const currentServicesGroupedByName: Record<string, CurrentService[]> = {};
	installs.forEach((container) => {
		let serviceContainerGroup = currentServicesGroupedByName[container.service_name];
		if (!serviceContainerGroup) {
			serviceContainerGroup = currentServicesGroupedByName[container.service_name] = [];
		}

		// remove the extra property that we added for the grouping
		delete container.service_name;
		serviceContainerGroup.push(container);
	});
```
since the ability to pass the accumulator forward is unused except to keep the original object in scope, which is easily done via it being a closure anyway